### PR TITLE
Maple Tree Generation

### PR DESCRIPTION
**Pull request by PotatoHax101**

### CODE REVIEW

File: Forest.cs [hunk 1]
```diff
@@ -61,15 +69,21 @@ private void DisplayForest(Tree[,] forest)
...
                     else if (forest[j, i].GetType() == typeof(Fir))
                     {
                         Console.ForegroundColor = ConsoleColor.DarkGreen;
-                        Console.Write("f", Console.ForegroundColor);
+                        Console.Write("f ", Console.ForegroundColor);
+                    }
+                    else if (forest[j, i].GetType() == typeof(Maple))
+                    {
+                        Console.ForegroundColor = ConsoleColor.DarkMagenta;
+                        Console.Write("m ", Console.ForegroundColor);
```
Comment (PotatoHax101) [inline Forest.cs:82]:
```suggestion
                        Console.Write("m", Console.ForegroundColor);
```

File: Forest.cs [hunk 2]
```diff
@@ -61,15 +69,21 @@ private void DisplayForest(Tree[,] forest)
...
                     {
                         Console.ForegroundColor = ConsoleColor.DarkCyan;
-                        Console.Write("s", Console.ForegroundColor);
+                        Console.Write("s ", Console.ForegroundColor);
                     }
                     else if (forest[j, i].GetType() == typeof(Fir))
                     {
                         Console.ForegroundColor = ConsoleColor.DarkGreen;
-                        Console.Write("f", Console.ForegroundColor);
+                        Console.Write("f ", Console.ForegroundColor);
```
Comment (PotatoHax101) [inline Forest.cs:77]:
```suggestion
                        Console.Write("f", Console.ForegroundColor);
```

File: Forest.cs [hunk 3]
```diff
@@ -61,15 +69,21 @@ private void DisplayForest(Tree[,] forest)
                     if (forest[j, i].GetType() == typeof(Spruce))
                     {
                         Console.ForegroundColor = ConsoleColor.DarkCyan;
-                        Console.Write("s", Console.ForegroundColor);
+                        Console.Write("s ", Console.ForegroundColor);
```
Comment (PotatoHax101) [inline Forest.cs:72]:
```suggestion
                        Console.Write("s", Console.ForegroundColor);
```

File: Forest.cs [hunk 4]
```diff
@@ -24,27 +24,35 @@ public Forest(int ForestWidth, int ForestHeight)
                 // Row
                 for (int j = 0; j < forest.GetLength(0); j++)
                 {
-                    forest[j, i] = DecideTree(rng);
+                    forest[j, i] = DecideTree(rng, i + j + 1);
```
Comment (tomkettle1) [inline Forest.cs:27]: (As we've talked) this works, but is a lil confusing. Might need a comment

File: Forest.cs [hunk 5]
```diff
@@ -24,27 +24,35 @@ public Forest(int ForestWidth, int ForestHeight)
...
+
+            // If it's the third tree create maple
+            if ((index % 3) == 0)
+            {
+                outputTree = new Maple(0);
+                return outputTree;
+            }
+
+            // Ratio is 4:1 so getting random number out of 50
             int chosenNumber = rng.Next(50);
```
Comment (tomkettle1) [inline Forest.cs:47]: Should be 51 (or 5)

File: Forest.cs [hunk 6]
```diff
@@ -24,27 +24,35 @@ public Forest(int ForestWidth, int ForestHeight)
...
             if (chosenNumber <= 10)
```
Comment (tomkettle1) [inline Forest.cs:48]:
```suggestion
            int chosenNumber = rng.Next(1, 6);
            if (chosenNumber == 1)
```

Comment (tomkettle1) [inline Forest.cs:48]:
```suggestion
            int chosenNumber = rng.Next(1, 6);
            if (chosenNumber == 1)
```

**Review by tomkettle1 — CHANGES_REQUESTED** [review]
All looks good, except a few alterations (from a prev pull tho)

File: Forest.cs (hunk 4 above)
Comment (PotatoHax101) [reply Forest.cs:27]:
```suggestion
                    // Adds the indexes of the two dimensions of the array (+1)
                    // Gets how far into the 2D array an element is   
                    forest[j, i] = DecideTree(rng, i + j + 1);
```

File: Forest.cs [hunk 7]
```diff
@@ -24,27 +24,37 @@ public Forest(int ForestWidth, int ForestHeight)
                 // Row
                 for (int j = 0; j < forest.GetLength(0); j++)
                 {
-                    forest[j, i] = DecideTree(rng);
+                    // Adds the indexes of the two dimensions of the array (+1)
+                    // Gets how far into the 2D array an element is
```
Comment (tomkettle1) [inline Forest.cs:28]:
```suggestion
                    // Gets how far into the 2D array an element is
                    // Quick method of finding (forest.GetLength(0) * i) + j
```

**Review by tomkettle1 — DISMISSED** [review]
All good! Wait for @TimMcGilly to review tho

**Comment by tomkettle1** [comment]
WAIT. Actually I think the index function only works if its a multiple of 3 + 1 wide

**Pull request closed without merging**